### PR TITLE
Small things 10

### PR DESCRIPTION
**Pull request by Hornwitser**
More isolated tasks from my todo list. See individual commits for changes and rationale for the changes.

### CODE REVIEW

File: packages/lib/factorio/patch.js [hunk 1]
```diff
@@ -251,8 +251,14 @@ async function patch(savePath, modules) {
 	// Write back the save
 	let tempSavePath = `${savePath}.tmp`;
 	let stream = zip.generateNodeStream({ compression: "DEFLATE" });
-	let pipe = stream.pipe(fs.createWriteStream(tempSavePath));
-	await events.once(pipe, "finish");
+	let fd = await fs.open(tempSavePath, "w");
+	try {
+		let pipe = stream.pipe(fs.createWriteStream("", { fd, autoClose: false }));
+		await events.once(pipe, "finish");
+		await fs.fsync(fd);
```
Comment (psihius) [inline packages/lib/factorio/patch.js:258]: You don't need this line - the `finally` block is always executed.

**Review by psihius — APPROVED** [review]
Except for that one feedback comment, looks good.

File: packages/lib/factorio/patch.js (hunk 1 above)
Comment (Hornwitser) [reply packages/lib/factorio/patch.js:258]: I am confused by what you mean with this. Which line? The fsync one?

Comment (psihius) [reply packages/lib/factorio/patch.js:258]: @Hornwitser nevermind, i'm an idiot. Lack of sleep today made me thing both lines where a `.close()`

**Pull request merged**